File: lib/data/user_profile.dart
class UserProfile {
  final String uid; // this IS the E.164 phone number, per AuthManager.kt
  final String name;
  final String phone;
  final String country;
  final String photoUrl; // Base64-encoded JPEG string, NOT a URL (see PrivateChatScreen.kt note)
  final int createdAt;

  UserProfile({
    required this.uid,
    required this.name,
    required this.phone,
    required this.country,
    required this.photoUrl,
    required this.createdAt,
  });

  factory UserProfile.fromMap(Map<dynamic, dynamic> map) {
    return UserProfile(
      uid: map['uid'] as String? ?? '',
      name: map['name'] as String? ?? '',
      phone: map['phone'] as String? ?? '',
      country: map['country'] as String? ?? '',
      photoUrl: map['photoUrl'] as String? ?? '',
      createdAt: (map['createdAt'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'uid': uid,
      'name': name,
      'phone': phone,
      'country': country,
      'photoUrl': photoUrl,
      'createdAt': createdAt,
    };
  }
}

File: android/app/src/main/kotlin/com/tbtrapp/data/UserProfile.kt
package com.tbtrapp.data

// Mirrors lib/data/user_profile.dart field-for-field so a profile saved by
// either platform reads back identically on the other. `uid` IS the E.164
// phone number (see AuthManager.saveProfile) — there is no separate
// numeric user id.
//
// @JvmOverloads is required, not decorative: Firebase's
// snapshot.getValue(UserProfile::class.java) deserializes via reflection
// and needs a true zero-arg constructor. A Kotlin data class with only
// default-valued parameters does NOT generate one on its own — without
// @JvmOverloads every fetchProfile()/getValue() call would throw at
// runtime ("No matching constructor found").
data class UserProfile @JvmOverloads constructor(
    val uid: String = "",
    val name: String = "",
    val phone: String = "",
    val country: String = "",
    val photoUrl: String = "",
    val createdAt: Long = 0L
)
